calico_containers/calico_ctl/container.py

```diff
@@ -159,12 +159,12 @@ def container_add(container_id, ip, interface):
         # The ID is a path. Don't do any docker lookups
         workload_id = escape_etcd(container_id)
         orchestrator_id = NAMESPACE_ORCHESTRATOR_ID
-        namespace_path = container_id
+        namespace = netns.Namespace(container_id)
     else:
         info = get_container_info_or_exit(container_id)
         workload_id = info["Id"]
         orchestrator_id = DOCKER_ORCHESTRATOR_ID
-        namespace_path = "/proc/%s/ns/net" % info["State"]["Pid"]
+        namespace = netns.Namespace("/proc/%s/ns/net" % info["State"]["Pid"])
```

Comment:
Nope - that is the purpose of the PidNamespace class (or whatever it's called) ... it takes a PID as the argument.